media-libs/vidstab: new ebuild

thinrope opened the pull request:
Closes: https://bugs.gentoo.org/517892

This is a dependency for https://bugs.gentoo.org/517890 as well, not sure how can I mark it here.

Zlogene commented on media-libs/vidstab/vidstab-1.1.0.ebuild:
```diff
@@ -0,0 +1,23 @@
+MY_PN="vid.stab"
+DESCRIPTION="Video stabilization library"
+HOMEPAGE="http://public.hronopik.de/vid.stab/"
+SRC_URI="https://github.com/georgmartius/${MY_PN}/archive/v${PV}.tar.gz
```
one line please

thinrope replied:
fixed

Zlogene commented on media-libs/vidstab/vidstab-1.1.0.ebuild:
```diff
@@ -0,0 +1,23 @@
+IUSE=""
+
+RDEPEND=">=media-video/ffmpeg-3.2.4-r1"
+DEPEND="dev-util/cmake
```
do not depend on cmake

thinrope replied:
fixed (since inheriting cmake-utils does it, I guess)

Zlogene commented on media-libs/vidstab/metadata.xml:
```diff
@@ -0,0 +1,7 @@
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+  <maintainer type="person">
+    <email>maintainer-wanted@gentoo.org</email>
```
you dhpuld add yourself first if you are going to maintain it yourself first, and then add proxy-maintainers project last

thinrope replied:
I am not 100% sure I did it right, at least repoman does not complain.
BTW my Bugzilla account is different that my GitHub :-|

Zlogene commented on media-libs/vidstab/vidstab-9999.ebuild:
```diff
@@ -0,0 +1,20 @@
+
+EAPI=6
+
+inherit git-r3 cmake-utils
```
the same about cmake

thinrope replied:
I guess you meant the DEPEND? If so - I fixed it.

mgorny commented on media-libs/vidstab/vidstab-1.1.0.ebuild:
```diff
@@ -0,0 +1,21 @@
+MY_PN="vid.stab"
+DESCRIPTION="Video stabilization library"
+HOMEPAGE="http://public.hronopik.de/vid.stab/"
+SRC_URI="https://github.com/georgmartius/${MY_PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
```
Please use distfile name consistent with the directory inside it, i.e. `${MY_PN}-${PV}`. Also, please don't use `${MY_PN}` in github repo name — using full `https://github.com/georgmartius/vid.stab` will make it easier for people to copy-paste that into browser.

Both remarks considered, you can change `MY_PN` to `MY_P` and include version there.

thinrope replied:
I am not 100% sure I got it right, please have a look shortly.

mgorny commented on media-libs/vidstab/vidstab-1.1.0.ebuild:
```diff
@@ -0,0 +1,21 @@
+KEYWORDS="~amd64"
+IUSE=""
+
+RDEPEND=">=media-video/ffmpeg-3.2.4-r1"
```
Does it link to ffmpeg libraries? If so, please add `:=` to link the package to ffmpeg's SONAME and force rebuilding on ffmpeg ABI changes.

thinrope replied:
No it doesn't, ffmpeg uses it, not vice-versa:
$ ldd /usr/lib64/libvidstab.so.1.1 
	linux-vdso.so.1 (0x00007ffcdfd25000)
	libm.so.6 => /lib64/libm.so.6 (0x00007f5f513ca000)
	libgomp.so.1 => /usr/lib/gcc/x86_64-pc-linux-gnu/6.4.0/libgomp.so.1 (0x00007f5f5119c000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f5f50ded000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f5f518f6000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007f5f50be9000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f5f509c9000)
$ ldd /usr/bin/ffmpeg|grep vidstab
	libvidstab.so.1.1 => /usr/lib64/libvidstab.so.1.1 (0x00007fe69ebfc000)

mgorny replied:
Well, in that case the dependency should be in ffmpeg, and not here, shouldn't it?

thinrope replied:
Hmm... vidstab ist just a library, it cannot be run in a way, so you are right, should have no RDEPENDs - will fix.

mgorny commented on media-libs/vidstab/vidstab-9999.ebuild:
```diff
@@ -0,0 +1,19 @@
+DESCRIPTION="Video stabilization library"
+HOMEPAGE="http://public.hronopik.de/vid.stab/"
+EGIT_REPO_URI="https://github.com/georgmartius/vid.stab.git"
+EGIT_BRANCH="master"
```
Since master is the default branch, please don't specify it explicitly.

thinrope replied:
Ok, will fix shortly.

mgorny commented on media-libs/vidstab/vidstab-1.1.0.ebuild:
```diff
@@ -0,0 +1,21 @@
+HOMEPAGE="http://public.hronopik.de/vid.stab/"
+SRC_URI="https://github.com/georgmartius/${MY_PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
```
Are you sure it's not GPL-2+?

thinrope replied:
Yes indeed! Thanks for catching that, I missed the "version 2 , or later" part... will fix shortly.

mgorny commented on media-libs/vidstab/vidstab-1.1.0.ebuild:
```diff
@@ -0,0 +1,21 @@
+
+inherit cmake-utils
+
+MY_P="vidstab-${PV}"
```
I meant this to be `vid.stab-${PV}`, and you can reuse it in S then.

Zlogene commented:
Cold you plese squash multiple commits into one?

mgorny commented:
I wish such a nice simple ebuild would work. However, looking at CMakeLists.txt I see a few problems that you'll probably need to patch:
1. `add_definitions` adds `-O3` and `-g` which belong in user's CFLAGS, please strip that (patch or sed CMakeLists.txt).
2. `if(SSE2_FOUND)` forces SSE2 when detected on the host. That won't work when you build packages for another system. Apparently `SSE2_FOUND` is a cache variable, so you should be able to override its value. Add `USE=cpu_flags_x86_sse2` flag for that.
3. `USE_OMP` enables OpenMP by default. Please make that conditional over `USE=openmp`.
4. It would be nice to see if you could get the tests running. However, it looks like it's going to take some effort, so I'll understand if you don't want to give them a shot.

thinrope commented:
Thanks for all the precise feedback, a great learning experience so far! I'll do my best in the next days.

mgorny commented:
FYI, I've submitted a GNUInstallDirs PR upstream. However, you don't need to worry about that because we account for `LIB_SUFFIX` as well, so the old code works for us too.

candrews commented:
@thinrope would you mind creating a new PR and resuming work on this issue?

It appears this PR was getting close to acceptance when the Gentoo Github hack happened derailing it; it would be nice to get this work back.

thinrope commented:
@mgorny, I did fix 3/4 and submitted in a new PR #10797 as advised:
>     1. `add_definitions` adds `-O3` and `-g` which belong in user's CFLAGS, please strip that (patch or sed CMakeLists.txt).
> 
tried patching, but there are some changes in HEAD, so I needed different set of patches for 1.1.1 and 9999, so I ended using sed. So far seems reliable.

>     2. `if(SSE2_FOUND)` forces SSE2 when detected on the host. That won't work when you build packages for another system. Apparently `SSE2_FOUND` is a cache variable, so you should be able to override its value. Add `USE=cpu_flags_x86_sse2` flag for that.
> 
Implemented that.

>     3. `USE_OMP` enables OpenMP by default. Please make that conditional over `USE=openmp`.
> 
Implemented that.

>     4. It would be nice to see if you could get the tests running. However, it looks like it's going to take some effort, so I'll understand if you don't want to give them a shot.
>
will need some help, in the new PR #10797
